### ISSUE TITLE
null mixer not working

### ISSUE THREAD

**Issue by quantenschaum** [MusicPlayerDaemon/MPD#120]
The [doc](https://www.musicpd.org/doc/user/config_audio_outputs.html) says

 > the "null" mixer (null; allows setting the volume, but with no effect; this can be used as a trick to implement an external mixer)

So I would except, that I can set and get the volume from MPD, but playback is always at volume=100.

When I set `mixer_type "null"` in `mpd.conf` then I get

```
$ mpc volume 
volume: n/a
$ mpc volume 50
mpd error: problems setting volume
```

This is what `mixer_type "none"` should do, but not `null`. With this behaviour implementing an [external mixer](https://www.musicpd.org/doc/user/client_hacks.html#external_mixer) is impossible.

I have this issue on a Raspberry Pi running Raspbian Stretch and Music Player Daemon 0.19.21

**Comment by MaxKellermann** [comment]
Sorry, this ancient MPD version is unsupported.

**Comment by quantenschaum** [comment]
Ancient? Come on, it's from last year, not something like 0.14.

**Comment by MaxKellermann** [comment]
If you want to bitch around, take it to the Debian project. Maybe they provide support for old software they're shipping in the name of stability, for some definition of "stability".

**Comment by quantenschaum** [comment]
I just compiled Music Player Daemon 0.20.10. The null mixer behaves the same there, too.

**Comment by quantenschaum** [comment]
As far as I understand the docs, the null mixer is broken.

### Workaround 
1. Set `mixer_type "null"` or `mixer_type "none"`.
1. Create an additional dummy output with software mixer 
    ```
    audio_output {
     name "dummy"
     type "null"
     mixer_type "software"
    }
    ```

With the dummy output enabled `mpc volume` returns a valid value.

**Comment by quantenschaum** [comment]
Hey @MaxKellermann, instead of bitching around with sarcasm and just closing the issue, you should take a look at it. The current release is affect, too. And instead complaining about Debian's notion of stability (which I also find a bit strange), you better elaborate on the null mixer and how to use it correctly. I did RTFM and this is what puzzles me.

**Comment by quantenschaum** [comment]
Nothing? According to the docs "null allows setting the volume". With a single audio_output in mpd.conf using a null mixer, setting the volume is not possible. Tested with MPD 0.20.10. 
Please reopen this issue.

**Comment by tremby** [comment]
The way I read the docs, if you want to use a null volume control, you should set `mixer_type` to `"null"`. You seem to be setting it to `"software"`.

**Comment by tremby** [comment]
Oh, never mind; I misread your message. Deleting it and this. (This one is for the benefit of your email.)

**Comment by tremby** [comment]
I just gave it a try and setting `mixer_type` to `null` works for me for both `fifo` and `null` outputs.

**Comment by quantenschaum** [comment]
Yes, that was bit of confusion and was solved in #138.

Just to mention, I needed the `null` mixer use an external mixen to solve https://github.com/larsimmisch/pyalsaaudio/issues/8.
